operator/pkg/gateway-api/cell.go

```diff
@@ -178,3 +188,14 @@ func registerGatewayAPITypesToScheme(scheme *runtime.Scheme) error {
 
 	return nil
 }
+
+func registerAdditionalAPITypesToScheme(clientset k8sClient.Clientset, scheme *runtime.Scheme) error {
+	serviceImportSupport := checkCRD(context.Background(), clientset, mcsapiv1alpha1.SchemeGroupVersion.WithKind("serviceimports")) == nil
+	log.WithField("enabled", serviceImportSupport).
+		Info("Multi-cluster Service API ServiceImport GatewayAPI integration")
+	if serviceImportSupport {
+		return mcsapiv1alpha1.AddToScheme(scheme)
```

Comment:
nit: the name of this function intents that there could be multiple types that are registered. Hence, an early return might not be the best idea here. this would skip types that would be added below -> either removing the early return or rename the function to `registerMCSAPITypesToScheme` (preferred)